_posts/2020-03-13-password-protected-efk-stack-on-k8s.md

```diff
@@ -0,0 +1,436 @@
+---
+layout: post
+title: Password Protected EFK Stack on Kubernetes
+author: onuryilmaz
+---
+
+In this article, my aim is to play around with EFK stack on Kubernetes which is a collection of [Elasticsearch](https://www.elastic.co/elasticsearch/), [Fluentd](https://www.fluentd.org) and [Kibana](https://www.elastic.co/kibana/). The main motivation behind that stack is reliably and securely take the data from the k8s cluster, in any format, then search, analyze and visualize it any time. In brief,
+
+- **Elasticsearch** is a distributed, open source search and analytics engine for all types of data, including textual, numerical, geospatial, structured, and unstructured.
+- **Fluentd** is an open source data collector, which lets you unify the data collection and consumption for a better use and understanding of data.
+- **Kibana** is an open source frontend application providing search and data visualization capabilities for data indexed in Elasticsearch.
+
+I always prefer minikube for such experimental works, which makes easy to run k8s locally. Please note that minikube uses 1GB of memory and 2 CPU cores by default and these values are not sufficient for our EFK stack. You should start/configure minikube with the following command:
+
+``` yml
+$ minikube start --memory 8192 --cpus 4
+```
+
+If you have installed minikube successfully and run it with above configuration, you're about halfway through the EFK stack :)
+
+---
+
+To get started, create a namespace inside minikube, to do that you can simply run the following command:
+
+``` yml
+$ kubectl create ns kube-logging
+```
+
+A minor detail, you're going to find a file inside each module named as _kustomization.yaml_, [kustomize](https://kustomize.io) is another facilitator which lets us customize raw, template-free YAML files for multiple purposes, leaving the original YAML untouched and usable as is.
+
+***
+
+## Elasticsearch
+
+Let's get started to walk through the details of Elasticsearch which consists of a Statefulset, service, persistent volume and ConfigMap. You can always have a quick look to get the component summary from its _kustomization.yaml_ file.
+Firstly, we will claim a persistent volume from k8s for our Elasticsearch Statefulset which will be used to store the data collected by Fluentd.
```

Comment:
And I guess you wanted to put the `pvc.yaml` content here somewhere?

Reply:
Exactly, solved.